src/main/java/org/apache/mesos/offer/ResourceRequirement.java

```diff
@@ -15,12 +20,103 @@
  * or an already determined value.
  */
 public class ResourceRequirement {
```

Comment:
VIP and Port information is not relevant to every `ResourceRequirement`.  A sub-class (or some abstraction) should be used to add the information only where relevant.